skrf/vectorFitting.py

```diff
@@ -806,10 +812,9 @@ def _pole_relocation(poles, freqs, freq_responses, weights_responses, fit_consta
 
         # direct QR of stacked matrices for linalg.qr() only works with numpy>=1.22.0
         # workaround for old numpy:
-        R = np.empty((n_responses, dim_k, n_cols_unused + n_cols_used))
+        #R = np.empty((n_responses, dim_k, n_cols_unused + n_cols_used))
         A_ri = np.hstack((A.real, A.imag))
-        for i in range(n_responses):
```

Comment:
I just realized you have already started to replace the loop. But can't you simply use the direct method (a few lines above), which is currently commented out?